config-ui/src/hooks/data-scope/useTransformationsManager.jsx

```diff
@@ -94,33 +94,38 @@ const getDefaultTransformations = (provider) => {
 const useTransformationsManager = () => {
   const [transformations, setTransformations] = useState({})
 
-  const generateKey = (
-    connectionProvider,
-    connectionId,
-    projectNameOrBoard
-  ) => {
+  /**
+   * @param {string} connectionProvider
+   * @param {string} connectionId
+   * @param {Entity} entity
+   * @return {string}
+   */
+  const generateKey = (connectionProvider, connectionId, entity) => {
     let key = `not-distinguished`
     switch (connectionProvider) {
       case Providers.GITHUB:
       case Providers.GITLAB:
       case Providers.JENKINS:
-        key = projectNameOrBoard?.id
+        key = entity?.id
```

Comment:
We should be using the accessory function instead, `entity.getConfiguredEntityId()`
```suggestion
        key = entity?.getConfiguredEntityId()
```